hyperapp.d.ts

```diff
@@ -0,0 +1,149 @@
+declare module "hyperapp" {
+  export interface VNodeData {
+    [key: string]: any
+  }
+
+  export interface VNode {
+    tag: string
+    data: VNodeData
+    children: string | VNode | Array<VNode>
+  }
+
+  export type viewFunction<State, Actions> = (
```

Comment:
It is requested into the Application interface 
```tsx
export interface Application<
    State,
    ActionsInterface extends Actions<State> | undefined,
    EventsInterface extends Partial<Events<State, ActionsInterface>> | undefined
  > extends Partial<Mixin<State, ActionsInterface, EventsInterface>> {
    root: HTMLElement
    view: viewFunction<State, ActionsInterface>
  }
```
and the render event 
```tsx
 export interface Events<
    State,
    ActionsInterface extends Actions<State> | undefined
  > {
 render:
      | (
          state: State,
          actions: ActionsInterface,
          view: viewFunction<State, ActionsInterface>,
          emit: emitFunction<any>
        ) => viewFunction<State, ActionsInterface>
  }
```

For `vnodeFunction` I wanted a way to express a function that return a vnode for `h`, and I wasnt sure about Component. But as the discution is closed, we can move into now.
```tsx
export function h(
    tag: string | vnodeFunction,
    data?: VNodeData,
    children?: Array<number | string | VNode>
  ): VNode
```